.github/workflows/test_periodic.yml

```diff
@@ -78,30 +70,45 @@ jobs:
         echo "GOPATH=$GOPATH" >> $GITHUB_ENV
         echo "$GOPATH/bin" >> $GITHUB_PATH
 
-    - name: Cache Kubernetes
-      id: cache-k8s
-      uses: actions/cache@v2
-      with:
-        path: ${{ env.GOPATH }}/src/k8s.io/kubernetes/
-        key: kubernetes-${{ env.K8S_VERSION }}
-
     - name: Set up Go
       uses: actions/setup-go@v1
       with:
         go-version: ${{ env.GO_VERSION }}
-      if: steps.cache-k8s.outputs.cache-hit != 'true'
 
-    - name: Build and install Kubernetes
-      if: steps.cache-k8s.outputs.cache-hit != 'true'
+    - name: Install KIND
+      run: |
+        sudo curl -Lo /usr/local/bin/kind https://github.com/aojea/kind/releases/download/dualstack/kind
+        sudo chmod +x /usr/local/bin/kind
+
+    - name: Clone Kubernetes
       run: |
         set -x
         rm -rf $GOPATH/src/k8s.io/kubernetes
         git clone --single-branch --branch $K8S_VERSION https://github.com/kubernetes/kubernetes.git $GOPATH/src/k8s.io/kubernetes/
+        cd $GOPATH/src/k8s.io/kubernetes
+        source hack/lib/version.sh
+        kube::version::get_version_vars
+        echo "KUBE_GIT_VERSION=$KUBE_GIT_VERSION" >> $GITHUB_ENV
+        
+    - name: Cache Kubernetes
+      id: cache-k8s
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.GOPATH }}/src/k8s.io/kubernetes
+        key: k8s-go-master-${{ env.KUBE_GIT_VERSION }}
```

Comment:
we need to cache based on the commit hash or we'll hit the cache and not compile again the code